tests/test_options.py

```diff
@@ -57,7 +59,7 @@ def test_service_name_from_kwargs(self):
     )
     def test_service_name_from_env(self):
         options = Options()
-        self.assertIsInstance(options.resource_attributes, dict)
```

Comment:
I think that a test for `OTEL_SERVICE_NAME` env var should be added.
We can consider adding one for `SPLUNK_SERVICE_NAME` as well.

I think making a parametrized test would make it easier e.g. using `subtest`. reference: https://stackoverflow.com/a/34094/859275

Reply:
We have a test for this in OpenTelemetry Python and it is a supported feature in 1.2+. I feel testing what Otel guarantees to be stable functionality is very redundant. 

I can add a test for SPLUNK_SERVICE_NAME backward compatibility though.

Reply:
Not needed

Reply:
Added test for backward compat check.

Reply:
Actually, I added a test for it since it was very cheap. It should at least let us know we broke this if we start creating custom resource classes without processing env var. Seems unlikely but possible and the solution is very cheap so why not?

Thanks